src/NServiceBus.Core.Tests/Sagas/ComponentTests/SagaPersisterTests.cs

```diff
@@ -0,0 +1,208 @@
+        protected Task<TSagaData> GetByIdAndUpdate(Guid sagaId, Action<TSagaData> update, params Type[] availableTypes) => GetByIdAndUpdate<TSaga, TSagaData>(sagaId, update, availableTypes);
+        protected Task<TSagaData> GetByCorrelationPropertyAndUpdate(string correlatedPropertyName, object correlationPropertyData, Action<TSagaData> update) => GetByCorrelationPropertyAndUpdate<TSaga, TSagaData>(correlatedPropertyName, correlationPropertyData, update);
+        protected Task<TSagaData> GetByCorrelationPropertyAndComplete(string correlatedPropertyName, object correlationPropertyData) => GetByCorrelationPropertyAndComplete<TSaga, TSagaData>(correlatedPropertyName, correlationPropertyData);
+        protected Task<TSagaData> GetByCorrelationProperty(string correlatedPropertyName, object correlationPropertyData) => GetByCorrelationProperty<TSaga, TSagaData>(correlatedPropertyName, correlationPropertyData);
+        protected Task<TSagaData> GetById(Guid sagaId, params Type[] availableTypes) => GetById<TSaga, TSagaData>(sagaId, availableTypes);
+    }
+
+    public class SagaPersisterTests
+    {
+        protected PersistenceTestsConfiguration configuration;
+
+        [OneTimeSetUp]
+        public async Task OneTimeSetUp()
+        {
+            configuration = new PersistenceTestsConfiguration();
+            await configuration.Configure();
+        }
+
+        [OneTimeTearDown]
+        public async Task OneTimeTearDown()
+        {
+            await configuration.Cleanup();
+        }
+
+        protected async Task SaveSaga<TSaga, TSagaData>(TSagaData saga, params Type[] availableTypes)
+            where TSaga : Saga<TSagaData>, new()
+            where TSagaData : IContainSagaData, new()
+        {
+            var insertContextBag = configuration.GetContextBagForSagaStorage();
+            using (var insertSession = await configuration.SynchronizedStorage.OpenSession(insertContextBag))
+            {
+                var correlationProperty = SetActiveSagaInstanceForSave(insertContextBag, new TSaga(), saga, availableTypes);
+
+                await configuration.SagaStorage.Save(saga, correlationProperty, insertSession, insertContextBag);
+                await insertSession.CompleteAsync();
+            }
+        }
+
+        protected async Task<TSagaData> GetByIdAndComplete<TSaga, TSagaData>(Guid sagaId, params Type[] availableTypes)
+            where TSaga : Saga<TSagaData>, new()
+            where TSagaData : IContainSagaData, new()
+        {
+            var context = configuration.GetContextBagForSagaStorage();
+            TSagaData sagaData;
+            var persister = configuration.SagaStorage;
+            using (var completeSession = await configuration.SynchronizedStorage.OpenSession(context))
+            {
+                SetActiveSagaInstanceForGet<TSaga, TSagaData>(context, new TSagaData(), availableTypes);
```

Comment:
I think regardless of that, we need to find a way to make those components testable without essentially simulating the whole core pipeline. If the persistence requires information from the context it should set it itself instead of relying on core behavior?